src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudNoSnapshotCreatedIT.java

```diff
@@ -0,0 +1,139 @@
+  private static Logger log =
+      Logger.getLogger(ComputeEngineCloudNoSnapshotCreatedIT.class.getName());
+
+  private static final String SNAPSHOT_LABEL = "snapshot";
+  private static final int SNAPSHOT_TEST_TIMEOUT = 120;
+
+  @ClassRule public static Timeout timeout = new Timeout(5, TimeUnit.MINUTES);
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ByteArrayOutputStream logOutput = new ByteArrayOutputStream();
+  private static StreamHandler streamHandler = new StreamHandler(logOutput, new SimpleFormatter());
+  private static ComputeClient client;
+  private static Map<String, String> label = getLabel(ComputeEngineCloudNoSnapshotCreatedIT.class);
+  private static String name;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    ComputeEngineCloud cloud = initCloud(jenkinsRule);
+    handleClassLogs(streamHandler, ComputeEngineCloud.class.getName());
+    client = initClient(jenkinsRule, label, log);
+    handleClassLogs(streamHandler, ComputeClient.class.getName());
+
+    assertTrue(cloud.configurations.isEmpty());
+    InstanceConfiguration instanceConfiguration =
+        instanceConfiguration(
+            DEB_JAVA_STARTUP_SCRIPT,
+            NUM_EXECUTORS,
+            SNAPSHOT_LABEL,
+            label,
+            true,
+            true,
+            NULL_TEMPLATE);
+    cloud.addConfiguration(instanceConfiguration);
+    assertTrue(
+        logs(streamHandler, logOutput),
+        cloud.getInstanceConfig(instanceConfiguration.description).createSnapshot);
+
+    // Assert that there is 0 nodes
+    assertTrue(jenkinsRule.jenkins.getNodes().isEmpty());
+
+    FreeStyleProject project = jenkinsRule.createFreeStyleProject();
+    Builder step = new Shell("echo works");
+    project.getBuildersList().add(step);
+    project.setAssignedLabel(new LabelAtom(SNAPSHOT_LABEL));
+
+    FreeStyleBuild build = jenkinsRule.buildAndAssertSuccess(project);
+    Node worker = build.getBuiltOn();
+    assertNotNull(logs(streamHandler, logOutput), worker);
+    // Cannot handle class logs for ComputeEngineInstance until an instance exists.
+    handleClassLogs(streamHandler, ComputeEngineInstance.class.getName());
+
+    name = worker.getNodeName();
+
+    // Need time for one-shot instance to terminate and create the snapshot
+    Awaitility.await()
+        .timeout(SNAPSHOT_TEST_TIMEOUT, TimeUnit.SECONDS)
+        .until(() -> jenkinsRule.jenkins.getNode(worker.getNodeName()) == null);
+  }
+
+  @AfterClass
+  public static void teardown() throws IOException {
+    ITUtil.teardown(streamHandler, logOutput, client, label, log);
+  }
+
+  // Tests that no snapshot is created when we only have successful builds for given node
+  @Test
+  public void testNoSnapshotCreatedSnapshotNull() throws Exception {
+    assertNull(logs(streamHandler, logOutput), client.getSnapshot(PROJECT_ID, name));
```

Comment:
Suggest also adding logic verifying the state of the retrieved snapshot, potentially in its own test method.